internal/provider/resource_packer_channel_test.go

```diff
@@ -12,236 +12,243 @@ import (
+				ResourceName:      restrictedLatestConfig.ResourceName(),
+				ImportState:       true,
+				ImportStateId:     fmt.Sprintf("%s:%s", bucketSlug, channelSlug),
 				ImportStateVerify: true,
 			},
 		},
 	})
 }
 
-func TestAccPackerChannel_UpdateAssignedIteration(t *testing.T) {
-	resourceName := "hcp_packer_channel.production"
+func TestAccPackerChannel_RestrictionDrift(t *testing.T) {
+	bucketSlug := testAccCreateSlug("RestrictionDrift")
+	channelSlug := bucketSlug // No need for a different slug
+
+	channelUnrestrictedConfig := testAccPackerChannelBuilder("Drift", fmt.Sprintf("%q", channelSlug), fmt.Sprintf("%q", bucketSlug), "false")
+	channelRestrictedConfig := testAccPackerChannelBuilderFromChannel(channelUnrestrictedConfig, "true")
 
-	resource.Test(t, resource.TestCase{
-		PreCheck:          func() { testAccPreCheck(t, map[string]bool{"aws": false, "azure": false}) },
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t, map[string]bool{"aws": false, "azure": false})
+			upsertRegistry(t)
+			upsertBucket(t, bucketSlug)
+		},
 		ProviderFactories: providerFactories,
 		CheckDestroy: func(*terraform.State) error {
-			deleteBucket(t, acctestAlpineBucket, false)
+			deleteBucket(t, bucketSlug, true)
 			return nil
 		},
 		Steps: []resource.TestStep{
+			// Normal channels
 			{
+				Config: testConfig(testAccConfigBuildersToString(channelUnrestrictedConfig)),
+				Check:  testAccCheckPackerChannel(channelUnrestrictedConfig.ResourceName(), channelSlug, bucketSlug, "false"),
+			},
+			{ // Check drift mitigation for normal channel from false->true
 				PreConfig: func() {
-					fingerprint := "channel-update-it1"
-					upsertBucket(t, acctestAlpineBucket)
-					upsertIteration(t, acctestAlpineBucket, fingerprint)
-					itID, err := getIterationIDFromFingerPrint(t, acctestAlpineBucket, fingerprint)
-					if err != nil {
-						t.Fatal(err.Error())
-					}
-					upsertBuild(t, acctestAlpineBucket, fingerprint, itID)
+					updateChannelRestriction(t, bucketSlug, channelSlug, true)
 				},
-				Config: testConfig(testAccPackerChannelAssignedLatestIteration(acctestAlpineBucket, acctestProductionChannel)),
-				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttrSet(resourceName, "author_id"),
-					resource.TestCheckResourceAttrSet(resourceName, "created_at"),
-					resource.TestCheckResourceAttrSet(resourceName, "id"),
-					resource.TestCheckResourceAttr(resourceName, "bucket_name", acctestAlpineBucket),
-					resource.TestCheckResourceAttr(resourceName, "name", acctestProductionChannel),
-					resource.TestCheckResourceAttrSet(resourceName, "iteration.0.id"),
-					resource.TestCheckResourceAttr(resourceName, "iteration.0.fingerprint", "channel-update-it1"),
-				),
+				Config: testConfig(testAccConfigBuildersToString(channelUnrestrictedConfig)),
+				Check:  testAccCheckPackerChannel(channelUnrestrictedConfig.ResourceName(), channelSlug, bucketSlug, "false"),
```

Comment:
How is the drift mitigation bein tested in this case?

Reply:
The restriction is set to `true` OOB in the PreConfig step, and this step checks that after another `terraform apply`, the drift is mitigated by setting back to `false`